src/org/traccar/api/resource/CommandTypeResource.java

```diff
@@ -36,8 +36,12 @@
     @GET
     public Collection<CommandType> get(@QueryParam("deviceId") long deviceId,
             @QueryParam("textChannel") boolean textChannel) {
-        Context.getPermissionsManager().checkDevice(getUserId(), deviceId);
-        return Context.getDeviceManager().getCommandTypes(deviceId, textChannel);
+        if (deviceId != 0) {
+            Context.getPermissionsManager().checkDevice(getUserId(), deviceId);
+            return Context.getCommandsManager().getCommandTypes(deviceId, textChannel);
+        } else {
+            return Context.getCommandsManager().getAllCommandTypes();
```

Comment:
Allow get all supported by server command types (useful to create new Saved Command)